### PR TITLE
save genesis state to hot cache and do not look it up from disk

### PR DESCRIPTION
**Pull request by potuz**
Credit to @nisdas for finding this issue and to @kasey for solving it

Only lookup genesis root and then use the cache

### CODE REVIEW

**Review by terencechain — CHANGES_REQUESTED** [review]
I don't think this changes anything. Loading genesis block root results in the same genesis state anyway

**Comment by potuz** [comment]
> I don't think this changes anything. Loading genesis block root results in the same genesis state anyway

we do not want to use the disk

**Review by terencechain — APPROVED** [review]
Upon further clarification, this does not fix 0x0 vs finalized root issue but this allows loading by genesis state via cache which is nice to have. We need to ensure genesis state is saved

**Review by nisdas — APPROVED** [review]

File: beacon-chain/state/stategen/getter.go [hunk 1]
```diff
@@ -56,7 +56,11 @@ func (s *State) StateByRoot(ctx context.Context, blockRoot [32]byte) (state.Beac
 
 	// Genesis case. If block root is zero hash, short circuit to use genesis state stored in DB.
 	if blockRoot == params.BeaconConfig().ZeroHash {
-		return s.beaconDB.GenesisState(ctx)
+		root, err := s.beaconDB.GenesisBlockRoot(ctx)
```
Comment (qinlz2) [inline beacon-chain/state/stategen/getter.go:59]: Hello Author.

The change here is changing the semantics of the function. 

* New change is using `GenesisBlockRoot`: won't allow empty record
* Original code is using `GensisState`: allow empty record, and won't return error.  https://github.com/prysmaticlabs/prysm/blob/77d3ccb9ad8b28656452456d2d25437f5fc53d88/beacon-chain/db/kv/state.go#L91

I think this change will cause trouble when a fresh start of the chain (no `genesis-root` written yet)

Comment (nisdas) [reply beacon-chain/state/stategen/getter.go:59]: By right we should always have a genesis state, does this change break anything for you ?

Comment (qinlz2) [reply beacon-chain/state/stategen/getter.go:59]: - Yes, there is always a `genesis state`.   The new patch assume that this genesis state is stored in DB already. 

- Yes, could you try to bring up a "refresh" node?  Such as `interop`, it should prevent the node to start.

**Pull request merged**